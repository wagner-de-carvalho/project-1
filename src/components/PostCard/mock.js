export const postCardPropsMock = {
    title: 'Test title',
    body: 'Testing body',
    id: 1,
    cover: 'img/img.png'
};