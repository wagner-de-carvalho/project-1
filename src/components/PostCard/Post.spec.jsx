const { render, screen } = require("@testing-library/react")
import '@testing-library/jest-dom'
import { PostCard } from '.';
import { postCardPropsMock } from './mock';

const props = postCardPropsMock;

describe('<PostCard />', () => {
    it('should render POstCard correctly', () => {
        render(<PostCard {...props} />);

        expect(screen.getByRole('img', { name: 'Test title' }))
            .toHaveAttribute('src', 'img/img.png')

        expect(screen.getByRole('heading', { name: 'Test title' })).toBeInTheDocument()
        expect(screen.getByText('Testing body')).toBeInTheDocument()

    });

    it('should match snapshot', () => {
        const { container } = render(<PostCard {...props} />)
        expect(container.firstChild).toMatchSnapshot();
    })
})