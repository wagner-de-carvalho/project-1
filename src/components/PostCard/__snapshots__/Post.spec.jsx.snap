// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="Test title"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h2>
      Test title
    </h2>
    <p>
      Testing body
    </p>
  </div>
</div>
`;
